src/Main.elm

```diff
@@ -1,10 +1,226 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience exposing (Audience, AudienceType(..))
+import Data.AudienceFolder exposing (AudienceFolder)
+import Data.Mocked exposing (DirElem(..), Directory, parsedDirectory)
+import Html exposing (Attribute, Html, button, div, table, text, th, tr)
+import Html.Attributes exposing (classList, disabled)
+import Html.Events exposing (onClick)
+import List exposing (any, drop, filter, head, length, map, member)
+import Maybe.Extra exposing (join)
+import Result exposing (Result(..))
 
 
-main : Html msg
+type alias Model =
+    { directory : Directory
+    , parents : List (Maybe Int)
+    , filterBy : AudienceType
+    }
+
+
+type Msg
+    = OpenFolder AudienceFolder
+    | GoUp
+    | ChangeFilter AudienceType
+
+
+parentIs : Maybe Int -> DirElem -> Bool
+parentIs id de =
+    case de of
+        File audience ->
+            audience.folder == id
+
+        Folder audienceFolder ->
+            audienceFolder.parent == id
+
+
+typeIs : AudienceType -> DirElem -> Bool
+typeIs at de =
+    case ( de, at ) of
+        ( File audience, _ ) ->
+            audience.type_ == at
+
+        ( _, Shared ) ->
+            False
+
+        _ ->
+            True
```

Comment:
This is danger because of wildcard and also `True` here looks like bug. Nasty one, because when you add some new type in future, thx wildcard it will not popup in compiler. 
I see that this is meant to be only for File type check, but in that case I would prefer more explanatory name or keep this function next to File type definition (in the same module)

Reply:
Oh thanks. It was only to check the type of the `Audience` but it's true that it can be a pain to debug in the future.

Thank you very much for your advices! 😄